src/main/java/hudson/remoting/Channel.java

```diff
@@ -1725,11 +1738,27 @@ public Channel channel() {
         }
 
         /**
+         * If the channel is null, return the cause of the channel termination.
+         */
+        public Exception cause() {
+            return  cause;
+        }
+
+        /**
+         * @see Channel#getName()
+         */
+        @Nonnull
+        public String name() {
+            return name;
+        }
+
+        /**
          * Clears the {@link #channel} to signify that the {@link Channel} has been closed and break any complex
          * object cycles that might prevent the full garbage collection of the channel's associated object tree.
          */
-        public void clear() {
-            channel = null;
+        public void clear(Exception cause) {
```

Comment:
API breakage. Please keep the original method as a Deprecated(?) one with null cause